frontend/packages/console-dynamic-plugin-sdk/src/coderefs/coderef-resolver.ts

```diff
@@ -126,3 +126,33 @@ export const resolveCodeRefProperties = async <E extends Extension<P>, P = Exten
 
   return resolvedValues;
 };
+
+/**
+ *  Returns an extension with its resolved `CodeRef` properties.
+ */
+export const resolveExtensionWithCodeRefs = <E extends Extension<P>, P = ExtensionProperties<E>>(
+  ext: E,
+): Promise<Extension<ResolvedCodeRefProperties<P>>> => {
+  const refProperties = filterExecutableCodeRefProperties(ext.properties);
+  const refPromises = Object.values(refProperties).map((ref) => ref());
+
+  return Promise.all(refPromises).then((properties) => {
+    const resolvedRefs: ResolvedCodeRefProperties<P> = Object.assign(
```

Comment:
The code to transform coderef-only properties object (values as executable functions => values as resolved objects) seems kinda confusing to me.

Why not reuse existing function above? (basically an adaptation of existing `useResolvedExtensions` code)

```ts
const resolvedProperties = await resolveCodeRefProperties<E, P>(extension);
return Object.freeze(_.merge({}, extension, { properties: resolvedProperties })) as ResolvedExtension<E>;
```

This would make this function a two-liner.

Reply:
again, `resolveCodeRefProperties` awaits for the resolved value.
this causes render issues at least when trying to consume extensions outside a component (in our case redux).

I want to resolve all the properies in `Promise.all`, thats why we don't need `async` either

Reply:
We really should be looking to always consume extensions during the render cycle. Which also means setting up redux inside the render cycle instead of statically.